### ISSUE TITLE
Button without accessible name

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#3184]
## Description
The button that expands and collapses the "inbox" example on https://clarity.design/documentation/badges does not have an accessible name. Assistive technology users will not be able to identify this button or understand its purpose. 

See the button in question below.

<button class="clr-treenode-caret ng-tns-c2-0 ng-star-inserted" type="button" aria-expanded="false"><clr-icon class="clr-treenode-caret-icon" shape="caret" dir="right"><svg version="1.1" viewBox="0 0 36 36" preserveAspectRatio="xMidYMid meet" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" focusable="false" aria-hidden="true" role="img"><path class="clr-i-outline clr-i-outline-path-1" d="M29.52,22.52,18,10.6,6.48,22.52a1.7,1.7,0,0,0,2.45,2.36L18,15.49l9.08,9.39a1.7,1.7,0,0,0,2.45-2.36Z"></path></svg></clr-icon></button>

## Location
https://clarity.design/documentation/badges

## Priority
High

## Recommended fix
Use aria-label to provide an accessible name for this control. Provide scripting to update the  aria-label 
value according to the value of aria-expanded as seen below. 

```
<!-- when collapsed, the label should contain the word "expand" -->
<button aria-label="expand inbox" class="clr-treenode-caret ng-tns-c2-0 ng-star-inserted" type="button" aria-expanded="false">
```

```
<!-- when expanded, the label should contain the word "collapse"-->
<button aria-label="collapse inbox" class="clr-treenode-caret ng-tns-c2-0 ng-star-inserted" type="button" aria-expanded="true">
```

## LOE
Medium

**Comment by coryrylan** [comment]
fixed in https://github.com/vmware/clarity/pull/3206
